oneview/data_source_interconnect_type.go

```diff
@@ -0,0 +1,199 @@
+// (C) Copyright 2016 Hewlett Packard Enterprise Development LP
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// You may not use this file except in compliance with the License.
+// You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+package oneview
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceInterconnectType() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceInterconnectTypeRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"interconnect_bay_set": {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"interconnect_capabilities": {
+				Computed: true,
+				Type:     schema.TypeSet,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"max_bandwidth_in_gbps": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"capabilities": {
+							Type: schema.TypeSet,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+							Computed: true,
+						},
+					},
+				},
+			},
+			"downlink_port_capability": {
+				Computed: true,
+				Type:     schema.TypeSet,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"created": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"max_bandwidth_in_gbps": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"state": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"total_sub_port": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"uri": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"port_info": {
+				Optional: true,
+				Type:     schema.TypeList,
```

Comment:
Can this be replaced with TypeSet? Is order of the port_info required?

Reply:
Done